runtime/roots.c

```diff
@@ -46,6 +46,9 @@ CAMLexport void (*caml_scan_roots_hook)(scanning_action, void* fdata, struct dom
 
 void caml_do_roots (scanning_action f, void* fdata, struct domain* d, int do_final_val)
 {
+  if (d->state->tls)
```

Comment:
Why is this check needed? Since `tls` is a root, it should never be `NULL`. A freshly initialised `tls` should be `[]`, whose representation is `Val_unit` (== `0x1`). Hence, this check will never fail.

Reply:
It was `NULL` at times possibly due to not initialising `tls`. Works now after the initialisation is done. Thanks for pointing that out!